### PR TITLE
Kkraune/doc summaries

### PR DESCRIPTION
**Pull request by kkraune**
- I left a few todos, so need another round, but let's review the current doc and plan further improvements in next PR

(sorry for the long PR - it could be OK to merge it and read as a whole and do higher-level comments - the diff is quite unreadable)

### CODE REVIEW

File: en/document-summaries.html [hunk 1]
```diff
@@ -6,117 +6,220 @@
...
-        summary title type string {
-            source: title
-        }
-        summary year type int {
-            source: year
-        }
-    }
+    <span class="pre-hilite">document-summary content {</span>
+        <span class="pre-hilite">summary artist type string {</span>
+            <span class="pre-hilite">source: artist</span>
```
Comment (geirst) [inline en/document-summaries.html:163]: The source is not needed when it is equal to the summary field name. Consider skipping to simplify the example.

File: en/document-summaries.html [hunk 2]
```diff
@@ -6,117 +6,220 @@
...
+        <span class="pre-hilite">}</span>
+        <span class="pre-hilite">summary album type string {</span>
+            <span class="pre-hilite">source: album</span>
```
Comment (geirst) [inline en/document-summaries.html:166]: The source is not needed when it is equal to the summary field name. Consider skipping to simplify the example.

File: en/document-summaries.html [hunk 3]
```diff
@@ -6,117 +6,220 @@
...
+{% endhighlight %}</pre>
+<p>
+  Query performance depends on the fields returned, see the <a href="#performance">performance</a> section.
+</p>
+
 
 
 
-<h2 id="defining-summary-sets-in-the-schema">Defining summary sets in the schema</h2>
+<h2 id="dynamic-summaries">Dynamic summaries</h2>
```
Comment (geirst) [inline en/document-summaries.html:102]: I think summary classes (#summary-classes) is more important than this section, and is a more natural continuation of the above YQL examples. Consider switching the order.

File: en/reference/default-result-format.html [hunk 4]
```diff
@@ -239,13 +215,59 @@
...
     <td>
-      <ul>
-        <li>The named document <a href="schema-reference.html#field">fields</a>,
-          this where fields of search documents go.
-          Fields without value are not rendered.</li>
-        <li><a href="schema-reference.html#summary-features">summaryfeatures</a></li>
-        <li><a href="schema-reference.html#match-features">matchfeatures</a></li>
-      </ul></td>
+      <p>
+        The named document (schema) <a href="schema-reference.html#field">fields</a>
```
Comment (geirst) [inline en/reference/default-result-format.html:219]:
```suggestion
        The named document (schema) <a href="schema-reference.html#field">fields</a>.
```

**Review by geirst — APPROVED** [review]
Overall looks good. Some comments to consider.

File: en/document-summaries.html (hunk 1 above)
Comment (kkraune) [reply en/document-summaries.html:163]: great!

File: en/document-summaries.html [hunk 5]
```diff
@@ -6,117 +6,220 @@
 ---
 
 <p>
-  Use document summaries to configure which fields to include in results.
-  The <em>default</em> summary contains <em>all</em> fields that
-  are possible to include in summaries; all other summaries will contain
-  a subset of the fields included in the <em>default</em> summary.
+  Use <a href="query-language.html">YQL</a> to select which fields to include in results:
```
[the remark this answers is not in the document]
Comment (kkraune) [reply en/document-summaries.html:9]: Good point! I will address the todo's and write this in a better intro in next PR

File: en/document-summaries.html (hunk 3 above)
Comment (kkraune) [reply en/document-summaries.html:102]: Yes, that will make the flow better! I am not 100% sure, though, as the dynamic part is a functionally important piece, maybe more so than the classes, which is more of an optimization.

As I will rewrite the intro, I will consider and most likely follow your suggestion in next PR. 

Thanks for great feedback!

**Comment by kkraune** [comment]
argh. forgot to push the changes based on the review. they will be in the next PR ...

**Pull request merged**